model/clusters_mgmt/v1/node_pool_type.model

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Representation of a node pool in a cluster.
+class NodePool {
+	// ID used to identify the cluster that this machinepool is attached to.
+	link Cluster Cluster
+
+	// The number of Machines (and Nodes) to create.
+	// Replicas and autoscaling cannot be used together.    
+	Replicas Integer
+
+	// Details for auto-scaling the machine pool.
+	// Replicas and autoscaling cannot be used together.    
+	Autoscaling NodePoolAutoscaling
+
+    // Specifies whether health checks should be enabled for machines in the NodePool.
+    AutoRepair Boolean
```

Comment:
indentation, 

This is the bane of my existance with this repo, the indentation between ide's and this `model` syntax